ide/projectui/src/org/netbeans/modules/project/ui/ProjectUtilities.java

```diff
@@ -483,31 +481,31 @@ public static boolean closeAllDocuments(Project[] projects, boolean notifyUI, St
             // store project's documents
             // loop all project being closed
             for (Map.Entry<Project,Set<String>> entry : urls4project.entrySet()) {
-                storeProjectOpenFiles(entry.getKey(), entry.getValue(), groupName);
+                storeProjectOpenFiles(entry.getKey(), new ArrayList<>(entry.getValue()), groupName);
             }
         }
         
         return urls4project != null;
     }
     
-    public static void storeProjectOpenFiles(Project p, Set<String> urls, String groupName) {
+    public static void storeProjectOpenFiles(Project p, List<String> urls, String groupName) {
         AuxiliaryConfiguration aux = ProjectUtils.getAuxiliaryConfiguration(p);
         
-        Set<String> openFileUrls = getOpenFilesUrls(p, groupName);
-        if(urls.isEmpty() && openFileUrls.isEmpty()) {
+        List<String> openFileUrls = getOpenFilesUrls(p, groupName);
+        if (urls.isEmpty() && openFileUrls.isEmpty()) {
             // was empty, stays empty, leave
             return;                
-        }        
-        if(urls.size() == openFileUrls.size()) {
+        }
+        // check if file list changed, order matters
```

Comment:
Why not just urls.equals(openFileUrls)?

That works now that the two are List<String> instead of Set<String>.

Reply:
good point - simplified the section